internal/pkg/client/patch/patch.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package patch
+
+import (
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/apimachinery/pkg/util/jsonmergepatch"
+	"k8s.io/apimachinery/pkg/util/mergepatch"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+// Patch object
+type Patch struct {
+	Type types.PatchType
+	Data []byte
+}
+
+var metadataAccessor = meta.NewAccessor()
+
+// GetClientSideApplyPatch updates a resource using the same approach as running `kubectl apply`.
+// The implementation here has been mostly extracted from the apply command: k8s.io/kubernetes/pkg/kubectl/cmd/apply.go
+func GetClientSideApplyPatch(current, desired runtime.Object) (Patch, error) {
+	var patch Patch
+	// Serialize the current configuration of the object.
+	currentb, err := runtime.Encode(unstructured.UnstructuredJSONScheme, current)
+	if err != nil {
+		return patch, fmt.Errorf("could not serialize current %v", current)
+	}
+	// Retrieve the last applied configuration of the object from the annotation.
+	last, err := GetLastApplied(current)
+	if err != nil {
+		return patch, fmt.Errorf("could not retrieve original configuration from %v", desired)
```

Comment:
how about `could not retrieve previously applied configuration from annotation on %v`

Reply:
done.